packages/runtime/container-runtime/src/opLifecycle/README.md

```diff
@@ -155,3 +172,100 @@ Notice that the sequence numbers don’t matter here, as all ops will be based o
+| Op 1                   Contents: +--------------------+-----------------+-----------------+-----------------+-----------------+ |
+| SeqNum: 1                        | Op 1               | Op 2            | Op 3            | Op 4            | Op 5            | |
+| Type: "groupedBatch"             | Contents: "abcde"  | Contents: empty | Contents: empty | Contents: empty | Contents: empty | |
+|                                  | Compression: 'lz4' |                 |                 |                 |                 | |
+|                                  +--------------------+-----------------+-----------------+-----------------+-----------------+ |
++---------------------------------------------------------------------------------------------------------------------------------+
+```
+
+Can produce the following chunks:
+
+```
++-------------------------------------------------+
+| Chunk 1/2    Contents: +----------------------+ |
+| SeqNum: 1              |  +-----------------+ | |
+|                        |  | Contents: "abc" | | |
+|                        |  +-----------------+ | |
+|                        +----------------------+ |
++-------------------------------------------------+
+```
+
+```
++--------------------------------------------------------------------------------------------------------------------------+
+| Chunk 2/2    Contents: +---------------------------------------------------------------------------------------------+ | |
+| SeqNum: 2              |  +----------------+-----------------+-----------------+-----------------+-----------------+ | | |
+|                        |  | Contents: "de" | Contents: empty | Contents: empty | Contents: empty | Contents: empty | | | |
+|                        |  +----------------+-----------------+-----------------+-----------------+-----------------+ | | |
+|                        +---------------------------------------------------------------------------------------------+ | |
++--------------------------------------------------------------------------------------------------------------------------+
+```
+
+-   Send to service
+-   Service acks ops sent
+-   Receive chunks from service
+-   Recompile to the grouped batch step
+
+Ungrouped batch:
+
+```
++--------------------+-----------------+-----------------+-----------------+-----------------+
+| Op 1               | Op 2            | Op 3            | Op 4            | Op 5            |
+| Contents: "abcde"  | Contents: empty | Contents: empty | Contents: empty | Contents: empty |
+| SeqNum: 2          | SeqNum: 2       | SeqNum: 2       | SeqNum: 2       | SeqNum: 2       |
+| ClientSeqNum: 1    | ClientSeqNum: 2 | ClientSeqNum: 3 | ClientSeqNum: 4 | ClientSeqNum: 5 |
+| Compression: 'lz4' |                 |                 |                 |                 |
++--------------------+-----------------+-----------------+-----------------+-----------------+
```

Comment:
These have SeqNum 2 but originally were sent with SeqNum 1. That's not a problem? The explanation of the grouped batching setting above says the batched messages are given the sequence number of the parent grouped message, which a few diagrams above, is 1, right? Chunking then causes seqnums 1 and 2 to appear but the grouped batch's original SeqNum is 1, so that's what these should have?

Reply:
There is no assigned `sequenceNumber` when sending ops. That number is assigned only by service. AKA runtime sends `IDocumentMessage`, service responds with `ISequenceDocumentMessage`.

When we get the chunks back, the `sequenceNumber` starts at that last chunk (seqNum 2 in this case) as we may have had other ops sequenced in between the other chunks. This is also shown in the `## How it works` section.